zerver/actions/realm_settings.py

```diff
@@ -486,3 +487,67 @@ def do_send_realm_reactivation_email(realm: Realm, *, acting_user: Optional[User
         language=language,
         context=context,
     )
+
+
+def do_send_realm_deactivation_email(realm: Realm, acting_user: Optional[UserProfile]) -> None:
+    # if realm is not deactivated, but changed the domain
+    if not acting_user:
+        return
+
+    event_time = timezone_now()
+    RealmAuditLog.objects.create(
+        realm=realm,
+        acting_user=acting_user,
+        event_type=RealmAuditLog.REALM_DEACTIVATION_EMAIL_SENT,
+        event_time=event_time,
+        extra_data=orjson.dumps(
+            {
+                RealmAuditLog.ROLE_COUNT: realm_user_count_by_role(realm),
+            }
+        ).decode(),
+    )
+    owners = realm.get_human_owner_users()
+    admins = realm.get_human_admin_users(include_realm_owners=False)
+    admin = admins[0]
+    context = {
+        "realm_uri": realm.uri,
+        "realm_name": realm.name,
+        "event_date": event_time.date(),
+        "acting_user_name": acting_user.full_name,
+        "self_hosted": realm.PLAN_TYPE_SELF_HOSTED,
+        "admin_user_mail": admin.delivery_email,
+    }
+    language = realm.default_language
+
+    deactivating_owner = None
+    other_owners = []
+    for owner in owners:
+        if owner == acting_user:
+            deactivating_owner = owner
+        else:
+            other_owners.append(owner)
+
+    # Send separate emails to the deactivating owner and the rest of the owners
+    if deactivating_owner:
```

Comment:
I think we can just send the ids of the owners. Then in the template, we can check, if the acting_user is the same as the user to which the email is being sent and customize text accordingly using conditionals.

Hence, no need to use if/else here.

This approach will reduce the number of templates created (hence code duplication).

Reply:
Can you give some path for this to implement?

Reply:
@prakhar1144 I tried your idea of passing ids of the owner, but thing is, how can we check in the template that, to whom this template gonna sent

Reply:
Hm, makes sense!

So, we can do something like this:
use `send_email` twice (once for the acting_user and another for other users) in the function and pass a context boolean variable which signifies `if the mail is being sent to the acting-user` or not.

depending on this variable, you can customize text in the template.
(we would still be able to reduce the number of templates with this approach)

Reply:
To implement this, there will be a need for two context objects, one with a boolean true for acting_user and one with a boolean false for acting_user.
This is a better approach than creating 6 new files. :)